### PR TITLE
Add support for variations for free downloads

### LINKED ISSUE #4: Add support for Price ID selection
Currently the plugin works by allowing a purchase of any product to get a free download, but it doesn't allow defining a price ID.

### PR DESCRIPTION
**Pull request by robincornett**
Fixes #4

### CODE REVIEW

**Review by ashleyfae — CHANGES_REQUESTED** [review]
I selected "Software Licensing - Variable Priced Plugin: Single Site", but after saving and refreshing, it looks like this:

Screenshot from 2020-09-23 13-13-27

**Comment by robincornett** [comment]
So this needs some more thinking for the front end use and validation. I think I need to hold off until #6 is resolved, because right now validation isn't working at all without it, but the validation is looking for a product ID, and this saves the product ID with the price ID appended, so the matches no longer match.

**Comment by ashleyfae** [comment]
Yeah I had that same thought, but didn't want to dig into it without #6. :+1:

**Review by mihaijoldis — APPROVED** [review]

File: edd-license-free-download.php [hunk 1]
```diff
@@ -155,7 +155,7 @@ public static function save_data( $post_id ) {
 		if ( ! empty( $_POST['edd_lfd_products'] ) && is_array( $_POST['edd_lfd_products'] ) ) {
 			foreach ( $_POST['edd_lfd_products'] as $product ) {
 				if ( ! empty( $product ) ) {
-					$products[] = absint( $product );
+					$products[] = esc_attr( $product );
```
Comment (ashleyfae) [inline edd-license-free-download.php:158]: Maybe `sanitize_text_field()` is better here?

Typically you want `sanitize_{x}` when _saving data_ and `esc_{x}` when _displaying_ data. We're saving here, so a sanitize option is better.

**Comment by ashleyfae** [comment]
## Tested

- [x] Same product, different variation cannot access.
- [x] Same product, same variation can access.
- [x] If Free License product doesn't specify a variation, someone who purchases a variation can access.

**Review by ashleyfae — APPROVED** [review]

**Pull request merged**